From f479935cda5eb38a4a4a9e909b6f4f82a0c7533d Mon Sep 17 00:00:00 2001
From: Christoph Rackwitz <christoph.rackwitz@gmail.com>
Date: Thu, 8 Apr 2021 12:59:06 +0200
Subject: [PATCH] fix for #19870

HAVE_QT and HAVE_WIN32UI can both be true at the same time
if HAVE_QT, window_w32.cpp is not included in the build, see CMakeLists.txt
---
 modules/highgui/src/window.cpp | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/modules/highgui/src/window.cpp b/modules/highgui/src/window.cpp
index 406871b7e8f8..d2cf1e1e4826 100644
--- a/modules/highgui/src/window.cpp
+++ b/modules/highgui/src/window.cpp
@@ -93,7 +93,9 @@ CV_IMPL void cvSetWindowProperty(const char* name, int prop_id, double prop_valu
     break;
 
     case cv::WND_PROP_VSYNC:
-        #if defined (HAVE_WIN32UI)
+        #if defined (HAVE_QT)
+            // nothing
+        #elif defined (HAVE_WIN32UI)
             cvSetPropVsync_W32(name, (prop_value != 0));
         #else
             // not implemented yet for other toolkits
@@ -191,7 +193,9 @@ CV_IMPL double cvGetWindowProperty(const char* name, int prop_id)
     break;
 
     case cv::WND_PROP_VSYNC:
-        #if defined (HAVE_WIN32UI)
+        #if defined (HAVE_QT)
+            return -1;
+        #elif defined (HAVE_WIN32UI)
             return cvGetPropVsync_W32(name);
         #else
             return -1;
